controllers/component_dependency_update_controller_test.go

```diff
@@ -402,14 +403,232 @@ var _ = Describe("Component nudge controller", func() {
+				failureCount := getRenovateFailedEventCount()
+				// check that renovate config was created
+				renovateConfigsCreated := len(getRenovateConfigMapList())
+				// check that renovate pipeline run was created
+				renovatePipelinesCreated := len(getRenovatePipelineRunList())
+				return renovatePipelinesCreated == 1 && renovateConfigsCreated == 2 && failureCount == 0
+			}, timeout, interval).WithTimeout(ensureTimeout).Should(BeTrue())
+		}
+
+		It("Test build performs nudge on success, only namespace wide renovate config provided", func() {
+			customConfigType := "json"
+			customConfigMapName := types.NamespacedName{Namespace: UserNamespace, Name: NamespaceWideRenovateConfigMapName}
+			automergeValue := true
+			automergeTypeValue := "pr"
+			platformAutomergeValue := false
+			ignoreTestsValue := true
+			commitMessagePrefixValue := "msg_prefix"
+			commitMessageSuffixValue := "msg_suffix"
+			imageBuiltFrom := "repo/sha"
+			commitBodyValue := fmt.Sprintf("Image created from '%s'\n\nSigned-off-by:", imageBuiltFrom)
+			fileMatchValue := []string{"cfile1", "cfile2"}
+			customConfigMapData := map[string]string{
+				RenovateConfigMapAutomergeKey:           "true",
+				RenovateConfigMapCommitMessagePrefixKey: commitMessagePrefixValue,
+				RenovateConfigMapCommitMessageSuffixKey: commitMessageSuffixValue,
+				RenovateConfigMapFileMatchKey:           "cfile1, cfile2",
+				RenovateConfigMapAutomergeTypeKey:       "pr",
+				RenovateConfigMapPlatformAutomergeKey:   "false",
+				RenovateConfigMapIgnoreTestsKey:         "true",
+			}
+			assertRenovateConfiMap(nil, nil, customConfigMapData, imageBuiltFrom)
+
+			renovateConfigMaps := getRenovateConfigMapList()
+			Expect(len(renovateConfigMaps)).Should(Equal(2))
+			for _, renovateConfig := range renovateConfigMaps {
+				if strings.HasPrefix(renovateConfig.ObjectMeta.Name, "renovate-pipeline") {
+					for key, val := range renovateConfig.Data {
+						renovateConfigObj := &RenovateConfig{}
+						err := json.Unmarshal([]byte(val), renovateConfigObj)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(renovateConfigObj.Automerge).Should(Equal(automergeValue))
+						Expect(renovateConfigObj.AutomergeType).Should(Equal(automergeTypeValue))
+						Expect(renovateConfigObj.PlatformAutomerge).Should(Equal(platformAutomergeValue))
+						Expect(renovateConfigObj.IgnoreTests).Should(Equal(ignoreTestsValue))
+						Expect(renovateConfigObj.PackageRules[1].CommitMessagePrefix).Should(Equal(commitMessagePrefixValue))
+						Expect(renovateConfigObj.PackageRules[1].CommitMessageSuffix).Should(Equal(commitMessageSuffixValue))
+						Expect(strings.Contains(renovateConfigObj.PackageRules[1].CommitBody, commitBodyValue)).Should(BeTrue())
+						Expect(renovateConfigObj.CustomManagers[0].FileMatch).Should(Equal(fileMatchValue))
+						Expect(strings.HasSuffix(key, customConfigType))
+					}
+					break
+				}
+			}
+			deleteConfigMap(customConfigMapName)
```

Comment:
Looks like it could be moved to the common func as well

Reply:
moved